app/src/main/res/values/strings.xml

```diff
@@ -183,13 +183,19 @@
 
     <string name="login">Login</string>
     <string name="logout">Logout</string>
+    <string name="log_in">Log in</string>
     <string name="username">Username</string>
     <string name="password">Password</string>
     <string name="server">Server</string>
     <string name="unlimited">Unlimited</string>
     <string name="up_next">Up next</string>
     <string name="x_selected">%1$d selected</string>
 
+    <string name="top_error_message">Whoops</string>
+    <string name="middle_error_message">Slow or no network connection</string>
+    <string name="bottom_error_message">Please check your network settings</string>
```

Comment:
These three should be more descriptive, maybe add networking error to the beginning or something.